### DISCUSSION TITLE
How to serve deno binary from deno compile ?

### DISCUSSION THREAD

**Discussion by romainPrignon** [denoland/deno#14926] [Q&A] [answered]
hi :)

I tried to compile a small mod.ts into a full binary (with deno compile)
I also add it to my git repo (./bin dir) and I follow  the publish guide: https://deno.land/manual@main/publishing
Unfortunately, the payload was too large to store it on denoland.

What's the proper way to serve a deno binary ?
Should I host it on an external source ? like s3 ?

**Comment by zachauten** [comment, marked as answer]
https://deno.land/x doesn't serve binaries, it serves modules (scripts). You only really need `deno compile` if you're writing an application that you want to be able to ship somewhere that may not have the right version of deno. If you're writing a library or a script that you want to be able to import or run via url, you don't need `compile`.
